File: predict.py
from rpy2.robjects import r
import os

import misc

def main():
    resolution = misc.getResolution()
    feature_path = "data/features/%s/" % resolution
    data_path = "data/r/%s/" % resolution
    
    r('feature_path <- "%s"' % feature_path)
    r('data_path <- "%s"' % data_path)
    
    r.source("predict.r")
    
    print r['model']
    print r['prediction']
    print r['y']
    

    
if __name__ == "__main__":
    main()

File: predict.r
load(paste(data_path, "model.Rdata", sep=""))
features <- read.csv(paste(feature_path, "features.csv", sep=""), header = FALSE)
x <- features[1:nrow(features),1:ncol(features) - 1]
y <- features[1:nrow(features),ncol(features)]
prediction <- predict(model, x)
jpeg(paste(data_path, "residuals.jpg", sep=""))
plot(y, prediction, ylab="Residuals", xlab = "Larvae Count", main="Fly Habitats")
abline(0, 0)
dev.off()